flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -712,21 +740,15 @@ protected void afterInvoke() throws Exception {
         LOG.debug("Finished task {}", getName());
         getCompletionFuture().exceptionally(unused -> null).join();
 
-        final CompletableFuture<Void> timersFinishedFuture = new CompletableFuture<>();
-
-        // close all operators in a chain effect way
-        operatorChain.finishOperators(actionExecutor);
-
```

Comment:
You are going to conflict with https://github.com/apache/flink/pull/16556/commits/1655ec94600fcc7bc9ae4722e5783041784510dd (not yet merged, but soon part of https://github.com/apache/flink/pull/16556)

Reply:
I'll have to deal with it :(